model_definitions/bf6a52b2-b595-4358-ac4f-24fb41a85c45/model_modules/training.R

```diff
@@ -32,9 +32,13 @@ train <- function(data_conf, model_conf, ...) {
     table <- tbl(con, data_conf$table)
 
     # Create dataframe from tibble, selecting the necessary columns and mutating integer64 to integers
+    is.integer64 <- function(x){
+            class(x)=="integer64"
+                }
+  
     data <- table %>% select(c("NumTimesPrg", "PlGlcConc", "BloodP", "SkinThick", "TwoHourSerIns", "BMI", "DiPedFunc", "Age", "HasDiabetes")) %>%
-      mutate(NumTimesPrg = as.integer(NumTimesPrg), PlGlcConc = as.integer(PlGlcConc), BloodP = as.integer(BloodP), SkinThick = as.integer(SkinThick), TwoHourSerIns = as.integer(TwoHourSerIns), HasDiabetes = as.integer(HasDiabetes)) %>%
-      as.data.frame()
+            mutate_if(is.integer64, as.integer) %>%
```

Comment:
sorry for the delay with this.. If you run this code - it won't apply it on all rows. Basically, it will result in only the first 100 rows of the table being read from TD. It prints a warning to say this.

Is there some other option to use this and read all rows?

Reply:
You're right @willfleury, it was previously using the mutate_if function, but getting only 100 rows. We did those changes to avoid using tdplyr's scoped variant of `mutate_if`. As in tdplyr's official documentation:

_mutate_if and transmute_if scoped variants apply the predicate on a subset of 100 values for each column before generating the SQL for the SELECT statement. This can prevent the mutated expressions from being applied on some columns if the predicate needs to look at more than 100 values of a column. This value is currently not adjustable._

available at https://docs.teradata.com/r/2PawNCsNx4~wW3LHnC4kPw/SrbFIj553T6nOM93sXIOuA

thanks anyway for noting it @jeancharlesR